### PR TITLE
Fix protobuf run dependency on old libignition-msgs5 Windows packages

### PR DESCRIPTION
**Pull request by traversaro**
Checklist

* [ ] Used a static YAML file for the patch if possible ([instructions](https://github.com/conda-forge/conda-forge-repodata-patches-feedstock/blob/main/recipe/patch_yaml/README.md)).
* [ ] Only wrote code directly into `generate_patch_json.py` if absolutely necessary.
* [ ] Ran `python show_diff.py` and posted the output as part of the PR.
* [ ] Modifications won't affect packages built in the future. 

<!-- Put any other comments or information here --!>

### CODE REVIEW

**Review by beckermr — CHANGES_REQUESTED** [review]
Please submit these patches as YAML.

File: recipe/gen_patch_json.py [hunk 1]
```diff
@@ -3286,6 +3286,24 @@ def _gen_new_index_per_key(repodata, subdir, index_key):
...
             _replace_pin("jax >=0.2.6", "jax >=0.2.6,<0.4.14", record["depends"], record)
 
+       # Some libignition-msgs5 windows packages have missing libprotobuf constraints
+       # see https://github.com/conda-forge/libignition-msgs1-feedstock/issues/30
+       if record_name == "libignition-msgs5" and subdir == "win-64" and record.get("timestamp", 0) < 1611218248868:
+           if fn == "libignition-msgs5-5.3.0-h13ae965_2.tar.bz2":
+               record["depends"].append("libprotobuf >=3.13.0.1,<3.14.0a0")
+           if fn == "libignition-msgs5-5.3.0-h21ff451_1.tar.bz2":
+               record["depends"].append("libprotobuf >=3.13.0,<3.14.0a0")
+           if fn == "libignition-msgs5-5.3.0-h21ff451_0.tar.bz2:
```
Comment (beckermr) [inline recipe/gen_patch_json.py:3296]:
```suggestion
           if fn == "libignition-msgs5-5.3.0-h21ff451_0.tar.bz2":
```

**Comment by traversaro** [comment]
> Please submit these patches as YAML.

Sure! I had no idea about the new YAML system and I just noticed it in the PR message! Thanks for setting it up, I will convert this to YAML.

**Comment by traversaro** [comment]
(I am currently with limited bandwith (not a metaphor, I have limited GB on my phone), so I opened this to mark down the logic necessary to encode this https://github.com/conda-forge/libignition-msgs1-feedstock/issues/30#issuecomment-1682551859 . I will iterate on this locally when I have more bandwith.

**Comment by traversaro** [comment]
Superseded by https://github.com/conda-forge/conda-forge-repodata-patches-feedstock/pull/506 .

**Pull request closed without merging**